Add XIP-47: Group Chat Permissions (#58)
xmtp/XIPs: XIPs/xip-47-group-chat-permissions.md

---
xip: 47
title: Group Chat Permissions
description: This proposal describes a simple, flexible, updatable group chat permissions system.
author: Cameron Voell (@cameronvoell), Eleanor Hofstedt (@eleanorhofstedt)

> discussions-to: TBD

Reviewer: ```suggestion
discussions-to: https://community.xmtp.org/t/xip-47-group-chat-permissions/651
```

The ability to update permissions is needed to address the use cases of initial group permission misconfiguration and the inevitable evolution of a group’s trust dynamic. For example, just because an online project group chat starts as a small group of well-intentioned contributors does not mean that the group may not evolve to have a larger variety of member trustworthiness and contributor types over time.

## Backward compatibility

In addition to adding the new **Permission Policies** and **Permission Options** and making **Permissions Updatable**, we will also make the permission system itself updateable in the following ways:

- `libxmtp` can be updated so existing groups can add new **Permission Policies** via the proto `map<string, MetadataPolicy> update_metadata_policy`. This means that if later on we want to add a new Permission Policy to groups, such as "The Ability to Mute Group Members," we can add it without breaking any existing groups.

> - Deprecation and upgrade of the Permissions or Metadata Extensions in an existing group's Group Context

Reviewer: ```suggestion
- Deprecation and update of the Permissions or Metadata Extensions in an existing group's Group Context
```

> - Deprecation and upgrade of the Permissions or Metadata Extensions in an existing group's Group Context

Reviewer: Addressed!

> - Existing groups will not be able to include an entirely new Permissions or Metadata Extension until all group members update to a new version of libxmtp and perform a "leaf node update" commit they updated their "supported capabilities."  

Reviewer: @cameronvoell I think there is a missing or extra word in the line above - I can't tell which. =) Might you be able to help, please? Thank you!

Author: ```suggestion
- Existing groups will not be able to include an entirely new Permissions or Metadata Extension until all group members update to a new version of libxmtp and perform a "leaf node update commit" that updates their "supported capabilities."  
```
